src/dmd/optimize.d

```diff
@@ -504,6 +504,17 @@ Expression Expression_optimize(Expression e, int result, bool keepLvalue)
                     {
                         eint = ei;
                     }
+                    else if (auto se = ep.e1.isSymOffExp())
+                    {
+                        if (!se.var.isReference() &&
+                            !se.var.isImportedSymbol() &&
+                            se.var.isDataseg() &&
+                            se.var.isCsymbol())
```

Comment:
In this situation I have to mentally load myself with all of the previous conditions and then at the end I see that it applies only to C symbols. Why not start with this condition(preferably put it in the outer if)? That way, it becomes clear that what follows applies to C land. This should be a general rule.

Reply:
I realized the isCsymbol() isn't necessary and removed it.